service/src/main/java/greencity/service/NotificationServiceImpl.java

```diff
@@ -23,15 +27,32 @@ public class NotificationServiceImpl implements NotificationService {
     private final UserRepo userRepository;
     private final ModelMapper modelMapper;
 
+
+    private User findUserById(Long id) {
+        return userRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("User not found with id: " + id));
+    }
+
     /**
      * Creates a notification when a new comment is added.
      *
      * @param dto DTO with info about the new comment notification
      */
     @Override
     public void createNotification(NotificationDto dto) {
-        Notification notification = modelMapper.convert(dto, Notification.class);
-        notification.setReceiver(findUserById(dto.receiver().getId());
+        Notification notification = NotificationMapper.mapToEntity(dto);
+        if (dto instanceof NewReplyNotificationDto replyDto) {
+            String notificationText = NotificationTextFormatter.formatNewReplyText(
+                    replyDto.comment().getAuthor().getName(),
+                    "news",
+                    replyDto.comment().getEcoNewsTitle(),
+                    replyDto.comment().getModifiedDate()
+            );
+            notification.setText(notificationText);
+        }
```

Comment:
could just add a text() method in NotificationDto and implement it in NewReplyNotificationDto (and others) like
```java
public record NewReplyNotificationDto(/* ... */) implements NotificationDto {
    @Override
    public String text() {
        return NotificationTextFormatter.formatNewReplyText(authorName, "news", ecoNewsTitle, modifiedDate);
    }
}
```

generally checks like `if (objectOfBaseType instanceof Subtype objectOfSubtype)` are a bad practice